perception/autoware_traffic_light_arbiter/test/test_node.cpp

```diff
@@ -0,0 +1,276 @@
+{
+  auto node_options = rclcpp::NodeOptions{};
+  const auto package_dir =
+    ament_index_cpp::get_package_share_directory("autoware_traffic_light_arbiter");
+  node_options.arguments(
+    {"--ros-args", "--params-file", package_dir + "/config/traffic_light_arbiter.param.yaml"});
+  return std::make_shared<TrafficLightArbiter>(node_options);
+}
+
+void generateMap(LaneletMapBin & vector_map_msg)
+{
+  const auto package_dir = ament_index_cpp::get_package_share_directory("autoware_test_utils");
+  lanelet::LaneletMapPtr vector_map_ptr =
+    autoware::test_utils::loadMap(package_dir + "/test_map/lanelet2_map.osm");
+  vector_map_msg.header.frame_id = "base_link";
+  lanelet::utils::conversion::toBinMsg(vector_map_ptr, &vector_map_msg);
+}
+
+void generatePerceptionMsg(TrafficSignalArray & perception_msg)
+{
+  // traffic_light_group_id 1: 1012
+  {
+    TrafficSignal traffic_light_groups;
+    traffic_light_groups.traffic_light_group_id = 1012;
+    // elements 1: red + circle
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::RED;
+      elements.shape = TrafficElement::CIRCLE;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.9;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    // elements 2: green + right arrow
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::GREEN;
+      elements.shape = TrafficElement::RIGHT_ARROW;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.8;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    perception_msg.traffic_light_groups.push_back(traffic_light_groups);
+  }
+}
+
+void generateExternalMsg(TrafficSignalArray & external_msg)
+{
+  // traffic_light_group_id 1: 1012
+  {
+    TrafficSignal traffic_light_groups;
+    traffic_light_groups.traffic_light_group_id = 1012;
+    // elements 1: green + circle
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::GREEN;
+      elements.shape = TrafficElement::CIRCLE;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.6;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    // elements 2: green + right arrow
+    {
+      TrafficElement elements;
+      elements.color = TrafficElement::GREEN;
+      elements.shape = TrafficElement::RIGHT_ARROW;
+      elements.status = TrafficElement::SOLID_ON;
+      elements.confidence = 0.5;
+      traffic_light_groups.elements.push_back(elements);
+    }
+    external_msg.traffic_light_groups.push_back(traffic_light_groups);
+  }
+}
+
+bool isMsgEqual(const TrafficSignalArray & input_msg, const TrafficSignalArray & gt_msg)
+{
+  // check number of groups
+  if (input_msg.traffic_light_groups.size() != gt_msg.traffic_light_groups.size()) {
+    return false;
+  }
+
+  for (size_t group_idx = 0; group_idx < input_msg.traffic_light_groups.size(); ++group_idx) {
```

Comment:
```suggestion
  for (std::size_t group_idx = 0; group_idx < input_msg.traffic_light_groups.size(); ++group_idx) {
```

Reply:
Thank you for your review. I fixed in https://github.com/autowarefoundation/autoware.universe/pull/8747/commits/58c928b160251fd8a374964eeb2ee3eeba9002ac